backend/src/main/java/de/neuefische/backend/service/TimerService.java

```diff
@@ -16,8 +16,23 @@ public TimerService(TimerRepo timerRepo) {
     }
 
     public PumpTimer getTimer() {
-       return timerRepo.findFirstByOrderById().orElseThrow(
-               ()-> new IllegalStateException("No timer in database"));
+        PumpTimer pumpTimer = getTimerFromRepo();
+        if (!pumpTimer.isDone()){
+            setPumpTimerAsDone();
+        }
+        return pumpTimer;
+    }
+
+    private void setPumpTimerAsDone() {
+        PumpTimer pumpTimer = getTimerFromRepo();
+        pumpTimer.setDone(true);
+        timerRepo.deleteAll();
+        timerRepo.save(pumpTimer);
+    }
+
+    private PumpTimer getTimerFromRepo() {
+        return timerRepo.findFirstByOrderById().orElseThrow(
```

Comment:
shouldn't there be a parameter when its called find....ById?

Reply:
Yeah it's a bit misleading...the timers are ordered by id and there is only one timer in the repo at any time. timerRepo.findAll.get(0) would do the same.